pkg/image/validation/os.go

```diff
@@ -217,3 +219,35 @@ func validateUnattended(def *image.Definition) []FailedValidation {
 
 	return failures
 }
+
+func validateRawConfig(def *image.Definition) []FailedValidation {
+	var failures []FailedValidation
+	isValidSize := regexp.MustCompile(`^([1-9]\d+|[1-9])+[MGT]`).MatchString
+
+	if def.OperatingSystem.RawConfiguration.DiskSize == "" {
+		return nil
+	}
+
+	if def.Image.ImageType != image.TypeRAW {
+		msg := fmt.Sprintf("The 'rawConfiguration/diskSize' field can only be used when 'image type' is '%s'.", image.TypeRAW)
```

Comment:
These messages have been using the actual name of the attribute to make things easier on the user, so 'imageType' instead of 'image type'.